syntax = "proto3";

message Region{
  int64 row1 =1;
  int64 column1=2;
  int64 row2=3;
  int64 column2=4;
}

message ImageParameters {
  bytes data = 1;      //图像
  uint32 width = 2;    //宽度
  uint32 height = 3;   //高度
  string encoding = 4; //格式
  int64 encoder_value = 5;//图像底部对应的编码器计数
  int64 defect_encoder_value=6; //缺陷对应的编码器计数
  string defect_type=7;  //图片的缺陷种类
  int64 is_blow=8;  //图片所在段是否剔除
  uint64 timestamp = 9;  //时间戳
  string filename=10;  //图像文件名
  repeated Region regions = 11;  //存储缺陷区域
}